freqtrade/persistence/models.py

```diff
@@ -577,18 +577,19 @@ def recalc_trade_from_orders(self):
 
         total_amount = 0.0
         total_stake = 0.0
-        for temp_order in self.orders:
-            if (temp_order.ft_is_open or
-                    (temp_order.ft_order_side != 'buy') or
-                    (temp_order.status not in NON_OPEN_EXCHANGE_STATES)):
+        for o in self.orders:
+            if (o.ft_is_open or
+                    (o.ft_order_side != 'buy') or
+                    (o.status not in NON_OPEN_EXCHANGE_STATES)):
                 continue
 
-            tmp_amount = temp_order.amount
-            if temp_order.filled is not None:
-                tmp_amount = temp_order.filled
-            if tmp_amount > 0.0 and temp_order.average is not None:
+            tmp_amount = o.amount
+            tmp_price = o.average or o.price or 0.0
```

Comment:
If you don't fall back to 0.0 - you can check for `None` and ignore that order - which is simpler / makes the intend clearer.
If the order didn't have a price, nor an average, it can't be considered.

by falling back to 0.0, you'd add a stake of 0.0 below - which is pointless (but kindof hides the intend you had to exclude that order).

Reply:
Agreed. Fix implemented.